File: Python/Program/BarangElektronik.py
from Produk import Produk

# class child dari Produk
class BarangElektronik(Produk):
    def __init__(self, kodeProduk, stok, harga,namaBarang, garansi, tahunRilis):
        super().__init__(kodeProduk, stok, harga)
        self.namaBarang = namaBarang
        self.garansi = garansi
        self.tahunRilis = tahunRilis

    def get_NamaBarang(self):
        return self.namaBarang

    def set_NamaBarang(self, namaBarang):
        self.namaBarang = namaBarang

    def get_garansi(self):
        return self.garansi

    def set_garansi(self, garansi):
        self.garansi = garansi

    def get_TahunRilis(self):
        return self.tahunRilis

    def set_TahunRilis(self, tahunRilis):
        self.tahunRilis = tahunRilis

File: Java/Program/BarangElektronik.java
public class BarangElektronik extends Produk {
    private String namaBarang;
    private int garansi;     
    private int tahunRilis;

    public BarangElektronik(String kodeBarang, String namaBarang, int garansi, int harga, int stok, int tahunRilis) {
        super(kodeBarang, harga, stok);
        this.namaBarang = namaBarang;
        this.garansi = garansi;
        this.tahunRilis = tahunRilis;
    }

    public String getNamaBarang() {
        return namaBarang;
    }

    public void setNamaBarang(String namaBarang) {
        this.namaBarang = namaBarang;
    }

    public int getGaransi() {
        return garansi;
    }

    public void setGaransi(int garansi) {
        this.garansi = garansi;
    }

    public int getTahunRilis() {
        return tahunRilis;
    }

    public void setTahunRilis(int tahunRilis) {
        this.tahunRilis = tahunRilis;
    }
}
